src/Medidata.ZipkinTracer.Core.Collector/SpanProcessor.cs

```diff
@@ -53,71 +57,62 @@ public virtual void Stop()
 
         public virtual void Start()
         {
-            spanProcessorTaskFactory.CreateAndStart(() => LogSubmittedSpans());
+            spanProcessorTaskFactory.CreateAndStart(LogSubmittedSpans);
         }
 
         internal virtual void LogSubmittedSpans()
         {
             Span span;
-            spanQueue.TryTake(out span);
-            if (span != null)
+            while (spanQueue.TryTake(out span))
             {
-                logEntries.Add(Create(span));
+                serializableSpans.Enqueue(new SerializableSpan(span));
                 subsequentPollCount = 0;
             }
-            else if (logEntries.Any())
-            {
-                subsequentPollCount++;
-            }
+            if(serializableSpans.Count > 0) subsequentPollCount++;
 
-            if ((logEntries.Count() >= maxBatchSize)
-                || (logEntries.Any() && spanProcessorTaskFactory.IsTaskCancelled())
-                || (subsequentPollCount > MAX_NUMBER_OF_POLLS))
-            {
-                var entries = logEntries;
-                logEntries = new List<LogEntry>();
-                subsequentPollCount = 0;
-                Log(clientProvider, entries);
-            }
-        }
 
-        internal void Log(IClientProvider client, List<LogEntry> logEntries)
-        {
-            try
-            {
-                clientProvider.Log(logEntries);
-                retries = 0;
-            }
-            catch (TException)
+            if ((serializableSpans.Count() >= maxBatchSize)
+                || (serializableSpans.Any() && spanProcessorTaskFactory.IsTaskCancelled())
+                || (subsequentPollCount > MAX_NUMBER_OF_POLLS))
             {
-                if ( retries < 3 )
+                try
                 {
-                    retries++;
-                    Log(client, logEntries);
+                    if (serializableSpans.Any()) 
+                        SendSpansToZipkin(JsonConvert.SerializeObject(serializableSpans.ToList()));
```

Comment:
`serializableSpans` should be cleared and `subsequentPollCount ` is changed to `0` around here.